app/views/Export/ExportCodeInput.js

```diff
@@ -0,0 +1,235 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Alert,
+  KeyboardAvoidingView,
+  StatusBar,
+  StyleSheet,
+  TextInput,
+  View,
+} from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import { Icons } from '../../assets';
+import { Button } from '../../components/Button';
+import { IconButton } from '../../components/IconButton';
+import { Typography } from '../../components/Typography';
+import Colors from '../../constants/colors';
+import fontFamily from '../../constants/fonts';
+import { Theme } from '../../constants/themes';
+
+const CODE_LENGTH = 6;
+const MOCK_ENDPOINT =
+  'https://private-anon-da01e87e46-safeplaces.apiary-mock.com/access-code/valid';
+
+const CodeInput = ({ code, length, setCode }) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  let digits = [];
+  for (let i = 0; i < length; i++) digits.push(code[i]);
+
+  const digitRefs = useRef([]);
+  useEffect(() => {
+    digitRefs.current = digitRefs.current.slice(0, length);
+  }, [length]);
+
+  const focus = i => {
+    digitRefs.current[i].focus();
+  };
+
+  // Focus on mount
+  useEffect(() => {
+    setTimeout(() => {
+      focus(0);
+    }, 0); // allow waiting for transition to end & first paint
+  }, []);
+
+  const onFocus = i => {
+    if (i > currentIndex) {
+      // prohibit skipping forward
+      focus(currentIndex);
+    } else {
+      // restart at clicked digit
+      setCurrentIndex(i);
+      setCode(code.slice(0, i));
+    }
+  };
+
+  // Adding digits
+  const onChangeDigit = d => {
+    if (d.length) {
+      setCode(code.slice(0, currentIndex) + d);
+      const nextIndex = currentIndex + 1;
+      if (nextIndex < length) {
+        setCurrentIndex(nextIndex);
+        focus(nextIndex);
+      }
+    }
+  };
+
+  // Removing digits
+  const onKeyPress = e => {
+    if (e.nativeEvent.key === 'Backspace') {
+      // go to previous
+      if (!code[currentIndex]) {
+        const newIndex = currentIndex - 1;
+        if (newIndex >= 0) {
+          setCurrentIndex(newIndex);
+          setCode(code.slice(0, newIndex));
+          focus(newIndex);
+        }
+      }
+      // clear current (used for last digit)
+      else {
+        setCode(code.slice(0, currentIndex));
+      }
+    }
+  };
+
+  return (
+    <View style={{ flexDirection: 'row', flexShrink: 1 }}>
+      {[...digits].map((digit, i) => (
+        <TextInput
+          ref={ref => (digitRefs.current[i] = ref)}
+          key={`${i}CodeDigit`}
+          value={digit}
+          style={[
+            styles.digitInput,
+            {
+              borderColor: digit
```

Comment:
if digit is 0 it will be false. perhaps `digit >= 0`

Reply:
Misnomer on my part. It's really a character input, these are stored as a string for any generic character, so the evaluation is on existence. I updated the name to be all about `character` rather than `digit`